proposals/low-level-struct-improvements.md

```diff
@@ -1513,6 +1525,30 @@ void M(ref S s)
 
 This is correctly illegal in that case because the compiler has to consider the pathological case that `S.Data` could cyclic assign via `this`. That forces methods all methods that call `S.Data` to further mark their `ref` parameters as `[UnscopedRef]`. This is viral until the method which creates the value as a local. This is why *return only* exists as an escape scope. It does complicate the spec / implementation but it serves to make the feature significantly more usable.
 
+Note: this cyclic assignment problem does continue to exist for `[UnscopedRef] out` to `ref struct` because that causes the *safe-to-escape* and *ref-safe-to-escape* to be equivalent. 
+
+```c#
+ref struct RS
+{
+    int field;
+    ref int refField;
+}
+
+void M1(out RS p)
+{
+    // Error: method arguments must match rules do the following
+    // Step 1 calculate would calculate the smallest escape as *containing method*
+    // Step 2 would fail the assignment check because p safe-to-escape is *return only*
+    M1(out p);
+}
+
+void M2([UnscopedRef] out RS p)
```

Comment:
I gave a lot of consideration to `[UnscopedRef] out` being a compilation error. Basically just cut off the problem. But it is realistic, although very very corner, that someone would have assigned and returned an `out` to a `ref struct` by `ref` so decided to leave it.

Reply:
One thing I considered briefly was having the STE of `[UnscopedRef] out` be CallingMethod, to ensure that `outParam = M(ref outParam)` is an error. But it just started to feel too complicated. I'm happy with *play stupid games, win stupid prizes* as a resolution here.